Sources/tart/VMDirectory+OCI.swift

```diff
@@ -70,33 +74,61 @@ extension VMDirectory {
     let progress = Progress(totalUnitCount: diskCompressedSize)
     ProgressObserver(progress).log(defaultLogger)
 
+    //Create blobTmpDir and ProgressFile objects
+    let blobsDir = try BlobTmpDir(baseURL: baseURL)
+    let progFile = try ProgressFile(baseURL: baseURL)
+
+    //diskCount keeps track of diskLayer
+    var diskCount = 0
+
+    //loop pulls one blob at a time
     for diskLayer in diskLayers {
-      try await registry.pullBlob(diskLayer.digest) { data in
-        try filter.write(data)
-        progress.completedUnitCount += Int64(data.count)
+      diskCount += 1
+      if !progFile.isDiskLayerDownloaded(diskLayer: diskCount){
```

Comment:
1. Do we need a progress file if we can simply check for the existence of a blob file on the disk?

2. Can we build the `progFile` progress on the fly, without using additional state? Simply increment  the `completedUnitCount` on each successful layer download/cache hit, that's it.

Also, since we now we have two stages: downloading and extracting, it probably makes sense to rename the `progFile` into `progressDownload` and `progress` into `progressExtract` and not forget to use the `ProgressObserver` on them.